indico/modules/events/editing/__init__.py

```diff
@@ -81,6 +81,41 @@ def _get_cloners(sender, **kwargs):
     yield EditingSettingsCloner
 
 
+@signals.event.sidemenu.connect
+def _extend_event_menu(sender, **kwargs):
+    from indico.modules.events.editing.models.editable import EditableType
+    from indico.modules.events.layout.util import MenuEntryData
+
+    def _editing_visible(event, specific_type=None):
+        if not session.user or not event.has_feature('editing'):
+            return False
+
+        enabled_types = {
+            EditableType[et]
+            for et in event.editable_types
+            if (specific_type is None or specific_type.name == et)
+        }
+        if not enabled_types:
+            return False
+        elif event.can_manage(session.user, 'editing_manager'):
+            return True
+        return any(event.can_manage(session.user, et.editor_permission) for et in enabled_types)
```

Comment:
Does this need to be a clouse? Wouldn't it be better to keep it at the top level if it doesn't use any of the external scope vars (importing `EditableType` inside it shouldn't hurt)?

Reply:
No, doesn't *need* to be a closure, but I find it a bit ugly to define this globally in `__init__.py` even though it's completely local to the menu stuff. If we ever need it elsewhere I'd move it to a util though.

Reply:
Ah, it's because it's in `__init__.py`. You could always use `__all__`, but anyway, no big deal.